tests/plotting/test_renderer.py

```diff
@@ -176,4 +177,102 @@ def test_add_remove_legend(sphere):
 def test_legend_face(sphere, face):
     pl = pv.Plotter()
     pl.add_mesh(sphere, label='sphere')
-    pl.add_legend(face=face)
+    pl.add_legend(face=face, size=(0.5, 0.5))
+
+
+def test_legend_from_glyph(sphere):
+    pl = pv.Plotter()
+    x = sphere.face_normals[:, 0] ** 2
+    y = sphere.face_normals[:, 1] ** 2
+    z = sphere.face_normals[:, 2] ** 2
+
+    sphere['scale'] = (x**2 + y**2 + z**2) ** (1 / 2)
+    sphere['normals'] = sphere.face_normals * 0.1
+
+    arrows = sphere.glyph(scale='scale', orient='normals', tolerance=0.05)
+    pl.add_mesh(arrows, color='red', label='Magnitude')
+    pl.add_mesh(sphere)
+    pl.add_legend(size=(0.5, 0.5))
+
+
+def test_legend_from_multiple_glyph(random_hills):
+    pl = pv.Plotter()
+
+    random_hills["Normals2"] = -1 * random_hills["Normals"].copy()
+
+    arrows = random_hills.glyph(scale="Normals", orient="Normals", tolerance=0.05)
+    pl.add_mesh(arrows, color="black", label="label 1")
+
+    arrows2 = random_hills.glyph(scale="Normals", orient="Normals2", tolerance=0.05)
+    pl.add_mesh(arrows2, color="red", label="label 2")
+
+    pl.add_mesh(random_hills, scalars="Elevation", cmap="terrain", show_scalar_bar=False)
+
+    pl.add_legend(size=(0.5, 0.5))
+    pl.show()
```

Comment:
Should we have some tests that have image regression tests?  You would need to either put the test into `tests/plotting/test_plotting.py` where we automatically use the image regression test, or you need to `opt-in` through using `verify_image_cache` fixture for each test here.

Reply:
I don't mind either way.... for the moment I am going to pass the fixture.

Reply:
I added the new image cache to my last commit